# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 08925220d323
# Full Hash: 08925220d3230566245ba170f1a66e8c2a7d4a35
# Author: stransky <stransky@redhat.com>
# Date: 2024-08-09 21:27:00
# Description:
#   Bug 1912160 [Linux] Don't do drag drop after D&D operation end r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D218850
# ==============================================================================

diff -r 42ed64f52054 -r 08925220d323 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Fri Aug 09 14:50:41 2024 +0300
+++ b/widget/gtk/nsWindow.cpp	Fri Aug 09 11:48:28 2024 +0000
@@ -8860,6 +8860,10 @@
   RefPtr<nsDragService> dragService = nsDragService::GetInstance();
   nsDragSession* dragSession =
       static_cast<nsDragSession*>(dragService->GetCurrentSession(window));
+  if (!dragSession) {
+    LOGDRAG("    Received dragdrop after drag end.\n");
+    return FALSE;
+  }
   nsDragSession::AutoEventLoop loop(dragSession);
   return dragSession->ScheduleDropEvent(
       window, aDragContext, GetWindowDropPosition(window, aX, aY), aTime);
